File: 剑指offer/剑指 Offer 10- I. 斐波那契数列.cpp
/*
 * @Description: 
 * @Version: 1.0
 * @Author: Vicro
 * @Date: 2020-12-10 16:03:45
 * @LastEditTime: 2020-12-10 16:23:44
 * @FilePath: \Leetcode\剑指 Offer 10- I. 斐波那契数列.cpp
 */


/*
RESULT: Accept
TIME:     0ms    BEAT: 100.00%    O(n) = n
MEMORY: 6.2MB    BEAT:  25.71%    O(n) = 1
Description: 
*/

class Solution {
public:
    int fib(int n) {
        if (n <= 1) return n;
        int a = 0, b = 1, c = 0;

        for (int i = 2; i <= n; i ++) {
            c = (a + b) % 1000000007;
            a = b;
            b = c;
        }
        
        return c;
    }
};

File: 剑指offer/剑指 Offer 10- I. 斐波那契数列.py
'''
Description: 
Version: 1.0
Author: Vicro
Date: 2020-12-10 16:09:26
LastEditTime: 2020-12-10 16:11:56
FilePath: \Leetcode\剑指 Offer 10- I. 斐波那契数列.py
'''


"""
RESULT: Accept
TIME:     32ms    BEAT: 94.40%    O(n) = n
MEMORY: 13.5MB    BEAT: 15.20%    O(n) = 1
Description: 由于 Python 中整形数字的大小限制 取决计算机的内存 （可理解为无限大），因此可不考虑大数越界问题。
"""

class Solution:
    def fib(self, n):
        a, b = 0, 1
        
        for _ in range(n):
            a, b = b, a + b
        
        return a % 1000000007